File: 0205/Code/STUDY_FILES/study11/app01.py
# 변수
a = 1
print(a, type(a))

# 자료형
print(type(1))
print(type(1.0))
print(type(""))   # 편집과 특정 값 가져오기
print(type(True))
print(type(None))
print(type({}))
print(type([]))
print(type(()))

# 함수 생성
def fn():
  print("fn()")

# 함수 사용(요청)
fn()

# 클래스 선언
class Aclass:
  def __init__(self):
    pass
  def add(self, a, b):
    return a + b

# 클래스 생성(인스턴스 생성)
ac = Aclass()
print(ac.add(2,5))

# 제어문
# 조건문 (if, else, elif)
if 5 > 1: 
  print("5가 크다")
elif 5 < 1:
  pass
else:
  pass

# 반복문 (for) range(0) [] () ""
for i in "": # 값
  print(i)

arr = [1,2,3,4,5]
for i in range(len(arr)): # 인덱스 0, 1, 2, 3, 4
  print(i, arr[i])

arr = [[1,2],[1],[1,2,3]]
for y in range(len(arr)): # 인덱스
  for x in range(len(arr[y])): # 인덱스
    print(y, x, arr[y][x])

문자열 = "나는 한국인일까?"
print(문자열[:3])

arr1 = {}
arr2 = dict() # 형변환 함수
print(type(arr1), type(arr2))

print(type(int("1")))

print(10 == int("10")) # 참

obj = {}
obj["key"] = 10
print(obj, type(obj))

arr = [0]
arr[0] = 10
print(arr, type(arr))

arr = list()
arr.append(10)
print(arr, type(arr))
arr[0] = 20
print(arr, type(arr))

t = (1,2)
arr = list(t)
arr.append(10)
t = tuple(arr)
print(type(t), t)

arr2 = set(t)
arr3 = set(arr)
print(arr2 | arr3)

File: 0205/Code/STUDY_FILES/study11/app01.js
// 변수
var a = 1
let b = 1
const c = 1
console.log(a, b, c)

// 자료형
console.log(typeof 1)
console.log(typeof 1.0)
console.log(typeof "")  // 포멧팅 `{a}`
console.log(typeof true)
console.log(typeof null)
console.log(typeof undefined)
console.log(typeof {})
console.log(typeof [])

// 함수 생성
function fn1(){
  console.log("fn1()")
}
const fn2 = function(){
  console.log("fn2()")
}
const fn3 = () => {
  console.log("fn3()")
}

// 함수 사용(요청)
fn1()
fn2()
fn3()

// 클래스 선언
class Aclass {
  constructor(){}
  add(a, b){
    return a + b 
  }
}

// 클래스 생성(인스턴스 생성)
const ac = new Aclass()
console.log(ac.add(2,5))

// 제어문
// 조건문 (if, else, else if)

if(false) {
  console.log("5가 크다")
  console.log("5가 크다")
  console.log("5가 크다")
  console.log("5가 크다")
  console.log("5가 크다")
  console.log("5가 크다")
  console.log("5가 크다")
}
else if(5 < 1) {} 
else {}

// 반복문 (for)
for( ;false; ){
  console.log("반복")
}

for(const a in []){ // 인덱스
  
}
for(const a of []){ // 값

}

const arr = [[1,2],[1],[1,2,3]]
for(const y in arr) { // 인덱스
  for(const x in arr[y]) { // 인덱스
    console.log(y, x, arr[y][x])
  }
}

let 문자열 = "나는 한국인일까?"
console.log(문자열[-3])

const arr1 = new Object() // {}
const arr2 = {}
console.log(typeof arr1, typeof arr2)

console.log(typeof Number("1"))

console.log(10 === "10") // 참

obj = {}
obj["key"] = 10
console.log(obj, typeof obj)

const arr3 = []
arr3[0] = 10
console.log(arr3, typeof arr3)

const arr4 = new Array()
arr4.push(10)
console.log(arr4, typeof arr4)
arr4[0] = 20
console.log(arr4, typeof arr4)

